EIPS/eip-draft_deposit_contract_snapshot_interface.md

```diff
@@ -0,0 +1,282 @@
+---
+eip: <to be assigned>
+title: Deposit Contract Snapshot Interface
+description: Standardizing the format and endpoint for transmitting a snapshot of the deposit Merkle tree
+author: Mark Mackey (@ethDreamer)
+discussions-to: https://ethereum-magicians.org/t/URL-TO-BE-DECIDED-ONCE-EIP-NUMBER-ASSIGNED
+status: Draft
+type: Standards Track
+category: Interface
+created: 2021-01-29
+---
+
+## Abstract
+This EIP defines a standard format for transmitting the deposit contract Merkle tree in a compressed form during weak subjectivity sync. This allows newly syncing consensus clients to reconstruct the deposit tree much faster than downloading all historical deposits. The format proposed also allows clients to prune deposits that are no longer needed to participate fully in consensus (see [Deposit Finalization Flow](#deposit-finalization-flow)).
+
+## Motivation
+Most client implementations require beacon nodes to download and store every deposit log since the launch of the deposit contract in order to reconstruct the deposit Merkle tree. This approach requires nodes to store far more deposits than necessary to fully participate in consensus. It also needlessly increases the time it takes for new nodes to fully sync, which is especially noticable during weak subjectivity sync. Furthermore, if [EIP-4444](./eip-4444.md) is adopted, it will not always be possible to download all historical deposit logs from full nodes.
+
+This EIP solves these issues by defining a standard format for transmitting the deposit contract Merkle tree in a compressed form to newly syncing nodes during weak subjectivity sync.
+
+## Specification
+Consensus clients MAY continue to implement the deposit Merkle tree however they choose. However, when transmitting the tree to newly syncing nodes, clients MUST use the following format:
+
+```
+DepositTreeSnapshot {
+    finalized: Vector[Hash32],
+    deposits:  uint64,
+    eth1_block_hash: Hash32,
```

Comment:
```suggestion
    execution_block_hash: Hash32,
```
I'm open to other ideas, but we definitely shouldn't be calling it `eth1`.  Note: same variable name will need updated elsewhere in this document and associated assets.

Reply:
Done!